Remotion/Validation/Core/Validators/NotEmptyValidator.cs

```diff
@@ -1,62 +1,58 @@
-// Assembly location: C:\Development\Remotion\trunk-svn2git\packages\FluentValidation-Signed.5.0.0.1\lib\Net40\FluentValidation.dll
-
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
 using System;
 using System.Collections;
 using System.Linq;
+using JetBrains.Annotations;
 using Remotion.Validation.Implementation;
 
 namespace Remotion.Validation.Validators
 {
   public class NotEmptyValidator : PropertyValidator, INotEmptyValidator
   {
-    private readonly object _defaultValueForType;
-
-    public NotEmptyValidator (object defaultValueForType, IValidationMessage validationMessage = null)
+    public NotEmptyValidator ([CanBeNull] IValidationMessage validationMessage = null)
         : base (Constants.NotEmptyError, validationMessage ?? new InvariantValidationMessage (Constants.NotEmptyError))
     {
-      _defaultValueForType = defaultValueForType;
     }
 
     protected override bool IsValid (PropertyValidatorContext context)
     {
-      return context.PropertyValue != null
-             && !IsInvalidString (context.PropertyValue)
-             && !IsEmptyCollection (context.PropertyValue)
-             && !Equals (context.PropertyValue, _defaultValueForType);
-    }
+      var propertyValue = context.PropertyValue;
 
-    private bool IsEmptyCollection (object propertyValue)
-    {
-      if (propertyValue is IEnumerable source)
-        return !source.Cast<object>().Any();
+      if (propertyValue == null)
+        return true;
 
-      return false;
+      return !IsEmptyString (propertyValue) && !IsEmptyCollection (propertyValue);
     }
 
-    private bool IsInvalidString (object value)
+    private bool IsEmptyCollection (object propertyValue)
     {
-      if (value is string valueString)
-        return IsNullOrWhiteSpace (valueString);
+      if (propertyValue is IEnumerable enumerableValue)
+        return !enumerableValue.Cast<object>().Any();
 
       return false;
     }
 
-    private bool IsNullOrWhiteSpace (string value)
+    private bool IsEmptyString (object value)
     {
-      if (value == null)
-        return true;
+      if (value is string stringValue)
+        return stringValue == "" || stringValue.All (char.IsWhiteSpace);
```

Comment:
Empty gets reduced to vaild: Is-null || length > 0

Reply:
Could this be reduced to the following to cover both `IEnumerable` & `string`?
```csharp
if (propertyValue is IEnumerable enumerableValue)
  return !enumerableValue.Cast<object>().Any();
```

Reply:
Could yes. But it would be better in terms of cpu etc to do a test for ICollection (or maybe IReadOnlyCollection isntead) first. check what String Array and List<T> implement.